op/model.py

```diff
@@ -307,57 +314,39 @@ def __init__(self):
-            if use_json:
-                args += ('--format=json',)
-        else:
-            kwargs = dict()
-        result = run(args, check=True, **kwargs)
-        if capture_output:
-            text = result.stdout.decode('utf8')
-            if use_json:
-                return json.loads(text)
+    def _run(self, *args, return_output=True, use_json=True):
+        kwargs = dict(stdout=PIPE, stderr=PIPE)
+
+        if use_json:
+            args += ('--format=json',)
+
+        try:
+            result = run(args, check=True, **kwargs)
+        except CalledProcessError as e:
+            raise ModelError.from_exception(e)
+
+        if return_output:
+            if result.stdout is None:
+                return ''
             else:
-                return text
+                text = result.stdout.decode('utf8')
+                if use_json:
+                    return json.loads(text)
+                else:
+                    return text
 
     def relation_ids(self, relation_name):
         relation_ids = self._run('relation-ids', relation_name)
         return [int(relation_id.split(':')[-1]) for relation_id in relation_ids]
 
     def relation_list(self, relation_id):
-        try:
-            return self._run('relation-list', '-r', str(relation_id))
-        except CalledProcessError as e:
-            # TODO: This should use the return code if it is specific enough rather than the message.
-            # It seems to be 2 for this error, but I haven't been able to confirm yet if that might
-            # also apply to other error cases.
-            if b'relation not found' in e.stderr:
-                raise RelationNotFound() from e
-            else:
-                raise
+        return self._run('relation-list', '-r', str(relation_id))
 
     def relation_get(self, relation_id, member_name):
-        try:
-            return self._run('relation-get', '-r', str(relation_id), '-', member_name)
-        except CalledProcessError as e:
-            # TODO: This should use the return code if it is specific enough rather than the message.
-            # It seems to be 2 for this error, but I haven't been able to confirm yet if that might
-            # also apply to other error cases.
-            if b'relation not found' in e.stderr:
-                raise RelationNotFound() from e
-            else:
-                raise
+        return self._run('relation-get', '-r', str(relation_id), '-', member_name)
 
     def relation_set(self, relation_id, key, value):
-        try:
-            return self._run('relation-set', '-r', str(relation_id), f'{key}={value}', capture_output=False)
-        except CalledProcessError as e:
-            if b'relation not found' in e.stderr:
-                raise RelationNotFound() from e
-            else:
-                raise
+        return self._run('relation-set', '-r', str(relation_id), f'{key}={value}', return_output=False, use_json=False)
```

Comment:
How about we turn things the other way around, and have both of these false by default?  It'll probably be more natural to use it.